jgrapht-core/src/main/java/org/jgrapht/alg/AbstractStrongConnectivityInspector.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * (C) Copyright 2005-2017, by Christian Soltenborn and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.Vector;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.alg.interfaces.StrongConnectivityAlgorithm;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DirectedPseudograph;
+import org.jgrapht.graph.DirectedSubgraph;
+
+/**
+ * Base implementation of the strongly connected components algorithm.
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ *
+ * @author Christian Soltenborn
+ * @author Christian Hammer
+ * @author Dimitrios Michail
+ */
+abstract class AbstractStrongConnectivityInspector<V, E>
+    implements StrongConnectivityAlgorithm<V, E>
+{
+    protected final DirectedGraph<V, E> graph;
+    protected List<Set<V>> stronglyConnectedSets;
+    protected List<DirectedSubgraph<V, E>> stronglyConnectedSubgraphs;
+
+    public AbstractStrongConnectivityInspector(DirectedGraph<V, E> graph)
+    {
+        this.graph = Objects.requireNonNull(graph, "Graph cannot be null");
+    }
+
+    @Override
+    public DirectedGraph<V, E> getGraph()
+    {
+        return graph;
+    }
+
+    @Override
+    public boolean isStronglyConnected()
+    {
+        return stronglyConnectedSets().size() == 1;
+    }
+
+    @Override
+    public List<DirectedSubgraph<V, E>> stronglyConnectedSubgraphs()
+    {
+        if (stronglyConnectedSubgraphs == null) {
+            List<Set<V>> sets = stronglyConnectedSets();
+            stronglyConnectedSubgraphs = new Vector<>(sets.size());
+
+            for (Set<V> set : sets) {
+                stronglyConnectedSubgraphs.add(new DirectedSubgraph<>(graph, set, null));
+            }
+        }
+        return stronglyConnectedSubgraphs;
+    }
+
+    @Override
+    public DirectedGraph<DirectedSubgraph<V, E>, DefaultEdge> getCondensation()
```

Comment:
It seems that this implementation does not correspond exactly to the definition of a condensation. At least in the definitions I'm aware of, including the one on wikipedia, in a condensation there is only 1 arc from one component u to another component v. In this implementation you may have multiple arcs between u and v: one for every arc in the original graph that runs between component u and component v.

The only (limited) advantage that the current implementation offers, is that it can tell you how many edges are running between 2 different components, but it doesn't tell you which edges. However, since this feature isn't documented, it's unlikely that anyone will use it. 

To fix the issue, just replace `new DirectedPseudograph<>(DefaultEdge.class)` by `new SimpleDirectedGraph<>(DefaultEdge.class)`